packages/compass-e2e-tests/helpers/commands/disconnect.ts

```diff
@@ -15,76 +77,65 @@ export async function disconnectAll(
+  // bit, though.
+  const connectionItems = await browser.$$(
+    Selectors.Multiple.ConnectedConnectionItems
+  );
+  for (const connectionItem of connectionItems) {
+    const connectionName = await connectionItem.getAttribute(
+      'data-connection-name'
     );
-    // If we are still connecting, let's try cancelling the connection first
-    if (await cancelConnectionButtonElement.isDisplayed()) {
-      try {
-        await browser.closeConnectModal();
-      } catch (e) {
-        // If that failed, the button was probably gone before we managed to
-        // click it. Let's go through the whole disconnecting flow now
-      }
-    }
+    await browser.disconnectByName(connectionName);
+  }
 
-    await delay(100);
+  if (closeToasts) {
+    // If we disconnected "too soon" and we get an error like "Failed to
+    // retrieve server info" or similar, there might be an error or warning
+    // toast by now. If so, just close it otherwise the next test or connection
+    // attempt will be confused by it.
+    await browser.hideAllVisibleToasts();
   }
 
-  await browser.execute(() => {
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    require('electron').ipcRenderer.emit('app:disconnect');
-  });
+  // NOTE: unlike the single connection flow this doesn't make sure the New
+  // Connection modal is open after disconnecting.
+  // This also doesn't remove all connections from the sidebar so the
+  // connection will still be there, just disconnected.
+}
 
-  if (TEST_MULTIPLE_CONNECTIONS) {
-    // For multiple connections we're making the assumption that there should be
-    // no active connections left. Use a different command if you expect to
-    // disconnect just one connection and still keep others around.
-    await browser
-      .$(`${Selectors.SidebarTreeItems}[aria-expanded=true]`)
-      .waitForExist({ reverse: true });
-
-    // The potential problem here is that the list is virtual, so it is possible
-    // that not every connection is rendered and then this won't wait long
-    // enough. For now just wait an extra second just in case.
-    await browser.pause(1000);
-
-    if (closeToasts) {
-      // If we disconnected "too soon" and we get an error like "Failed to
-      // retrieve server info" or similar, there might be an error or warning
-      // toast by now. If so, just close it otherwise the next test or connection
-      // attempt will be confused by it.
-      if (await browser.$(Selectors.LGToastCloseButton).isExisting()) {
-        try {
-          const toastText = await browser.$('#lg-toast-region').getText();
-          debug('Closing toast', toastText);
-          await browser.clickVisible(Selectors.LGToastCloseButton);
-        } catch (error) {
-          debug('ignoring', error);
-        }
-      }
-    }
+export async function disconnectByName(
+  browser: CompassBrowser,
+  connectionName: string
+) {
+  await resetForDisconnect(browser, { closeToasts: false });
```

Comment:
if I get it right, the reset now happens 1 + N times for `disconnectAll`, maybe we could make it skippable in this case?

Reply:
doesn't it take some time?

Reply:
milliseconds. nothing compared to the general overhead of creating connections and connecting to them and browsing to where we need to go and waiting for toasts and for collections to load, etc, etc.